src/main/java/org/checkerframework/specimin/UnsolvedClassOrInterface.java

```diff
@@ -187,6 +187,50 @@ public Set<String> getClassFields() {
    * @param method the method to be added
    */
   public void addMethod(UnsolvedMethod method) {
+    // Check for another method with the same parameter list, but with differences in
+    // whether the parameter names are fully-qualified or simple names.
+    List<String> paramList = method.getParameterList();
+    UnsolvedMethod matchingMethod = null;
+    boolean preferOther = true;
+    methods:
+    for (UnsolvedMethod otherMethod : this.methods) {
+      // Skip methods that are definitely different.
+      if (!method.getName().equals(otherMethod.getName())
+          || otherMethod.getParameterList().size() != paramList.size()) {
+        continue;
+      }
+
+      for (int i = 0; i < paramList.size(); ++i) {
+        String paramType = paramList.get(i);
+        String otherParamType = otherMethod.getParameterList().get(i);
+        if ((this.packageName + "." + otherParamType).equals(paramType)) {
```

Comment:
Is `this.packageName` the package of the current class? How are we sure that it is also the package of the name of the related parameter?

Reply:
> Is `this.packageName` the package of the current class?

Yes. In particular, it's the name of the class that the method is being added to.

> How are we sure that it is also the package of the name of the related parameter?

We're not, I suppose. But we're relying on the fact that a simple name `Bar` used as a parameter in a class in package `com.foo` will be interpreted as `com.foo.Bar`.

Reply:
Thanks for the explanation.